models/org_team.go

```diff
@@ -34,6 +35,75 @@ type Team struct {
 	Units       []*TeamUnit `xorm:"-"`
 }
 
+// SearchTeamOptions holds the search options
+type SearchTeamOptions struct {
+	UserID      int64
+	Keyword     string
+	OrgID       int64
+	IncludeDesc bool
+	PageSize    int
+	Page        int
+}
+
+// SearchTeam search for teams. Caller is responsible to check permissions.
+func SearchTeam(opts *SearchTeamOptions) ([]*Team, int64, error) {
+	if opts.Page <= 0 {
+		opts.Page = 1
+	}
+	if opts.PageSize == 0 {
+		// Default limit
+		opts.PageSize = 10
+	}
+
+	var cond = builder.NewCond()
+
+	if len(opts.Keyword) > 0 {
+		lowerKeyword := strings.ToLower(opts.Keyword)
+		var keywordCond = builder.NewCond()
```

Comment:
var keywordCond = builder.Like{"lower_name"}
if opts.IncludeDesc {
     keywordCond = keywordCond.Or(builder.Like{"LOWER(description)", lowerKeyword})
}

Reply:
:(
`var keywordCond builder.Cond = builder.Like{"lower_name"}`

Reply:
Sorry, ran out of laptop battery before I could finish properly. Will fix later.

Reply:
Now I think I got it right.